### PR TITLE
Cucumber framework

### PR DESCRIPTION
**Pull request by plaird523**
Implemented Cucumber test framework, steps that were previously supported are now green

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[package]
+name = "partywithray-proof-of-membership"
+version = "0.1.0"
+edition = "2021"
```
Comment (neodaoist) [inline Cargo.toml:4]: 2022 ?

File: tests/cucumber.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+{
+    // Generate bindings for the 3rd party contracts we'll need
+
+
+    // Read fork endpoint from environment variable ETH_NODE_URL
+    let fork_endpoint = env::var("ETH_NODE_URL").expect("Environment variable ETH_NODE_URL should be defined and be a valid API URL");
+
+    let mut config: solidity_bdd::TestConfig = solidity_bdd::new();
+        config.fork(fork_endpoint, FORK_CHAIN_ID)
+        //.thirdPartyContracts([])  // FIXME: Add VRF here if we end up using it
```
Comment (neodaoist) [inline tests/cucumber.rs:32]: Nice nice, I like how you have this as an option.

File: tests/features/TheLow.feature [hunk 3]
```diff
@@ -8,9 +8,21 @@ Feature: Partywithray Proof of Membership
...
+        And each NFT description should be "TODODescriptionIncludeDesignerCredit"
+        And royalties should be set at 10% going to TODOBigNightENSName
+        And the ability to mint more NFTs should be frozen
+
+    Scenario: Mint on creation
+        Then the name should be "partywithray - The Low" and the symbol should "LOW"
+        And the supply should be 222
+        And all 222 NFTs should have the same art
+        And each NFT title should be "The Low {id}/222"
+            # Title uniqueness means we can't share metadata
```
Comment (neodaoist) [inline tests/features/TheLow.feature:25]: A clarifier, that depending on how we implement metadata, we maybe can share _most_ metadata for each tier. But I know what you mean generally.

**Review by neodaoist — APPROVED** [review]
LGTM

File: Cargo.toml (hunk 1 above)
Comment (plaird523) [reply Cargo.toml:4]: This is the Rust edition.  2015,2018, or 2021.  2021 is the latest.

File: tests/features/TheLow.feature (hunk 3 above)
Comment (plaird523) [reply tests/features/TheLow.feature:25]: We did discuss this.  I have it updated in another branch, actually.  Please ignore it here.

**Pull request merged**